apis/apis.go

```diff
@@ -97,7 +103,20 @@ func (a *AnalysisAPI) RunningAnalyses(ctx context.Context, username string, limi
 	return &data, nil
 }
 
+func (a *AnalysisAPI) RunningAnalysesAsync(ctx context.Context, itemsChan chan *AnalysisListing, errChan chan error, username string, limit int) {
+	analyses, err := a.RunningAnalyses(ctx, username, limit)
+	if err != nil {
+		errChan <- err
+		return
+	}
+	errChan <- nil
+	itemsChan <- analyses
+}
```

Comment:
This and all the `*Async` functions are basically the same. They take whatever they modify, add a `chan <normal return value>` and a `chan error` after the `ctx` argument, and then all the other arguments after that. Then, if it errors it sends that to the `chan error` or otherwise sends a nil to that channel and the actual return to the other channel.

That way, callers initialize the two channels, set it going, and later on can pull whatever off the error channel and determine if it failed before grabbing anything from the other channel.